libraries/plugins/social_network/include/golos/plugins/social_network/social_network.hpp

```diff
@@ -0,0 +1,193 @@
+#pragma once
+
+#include <appbase/application.hpp>
+#include <golos/plugins/chain/plugin.hpp>
+#include <golos/plugins/social_network/api_object/discussion.hpp>
+#include <golos/plugins/social_network/api_object/category_api_object.hpp>
+#include <golos/plugins/social_network/api_object/tag_api_object.hpp>
+
+namespace golos {
+    namespace plugins {
+        namespace social_network {
+            using plugins::json_rpc::msg_pack;
+
+            using get_tags_used_by_author_return_type = std::vector<std::pair<std::string, uint32_t>>;
+
+            DEFINE_API_ARGS(get_content,                           msg_pack, discussion)
+            DEFINE_API_ARGS(get_trending_tags,                     msg_pack, std::vector<tag_api_object>)
+            DEFINE_API_ARGS(get_content_replies,                   msg_pack, std::vector<discussion>)
+            DEFINE_API_ARGS(get_tags_used_by_author,               msg_pack, get_tags_used_by_author_return_type)
```

Comment:
Pack return parameters into structs, name them as in other plugins content_r, tags_used_by_author_r